File: remote/java/src/main/java/org/pxscene/rt/remote/RTRemoteTask.java
package org.pxscene.rt.remote;


import org.pxscene.rt.RTHelper;

/**
 * the rtRemote task, used to store the messages(include protocol) into task queue
 */
public class RTRemoteTask {


  /**
   * the protocol transport
   */
  private RTRemoteProtocol protocol;

  /**
   * the message from client
   */
  private RTRemoteMessage message;

  /**
   * create task with protocol and message
   *
   * @param protocol the protocol
   * @param message the message entity
   */
  public RTRemoteTask(RTRemoteProtocol protocol, RTRemoteMessage message) {
    this.protocol = protocol;
    this.message = message;
  }

  public RTRemoteProtocol getProtocol() {
    return this.protocol;
  }

  public void setProtocol(RTRemoteProtocol protocol) {
    RTHelper.ensureNotNull(protocol, "protocol");
    this.protocol = protocol;
  }

  public RTRemoteMessage getMessage() {
    return this.message;
  }

  public void setMessage(RTRemoteMessage message) {
    RTHelper.ensureNotNull(message, "message");
    this.message = message;
  }
}

File: remote/js/lib/RTRemoteTask.js
/**
 * Copyright (C) 2018 TopCoder Inc., All Rights Reserved.
 */

/**
 * the rtRemote task, used to store the messages(include protocol) into task queue
 *
 * @author      TCSCODER
 * @version     1.0
 */

/**
 * the rt remote task class
 */
class RTRemoteTask {
  /**
   * create new remote task
   * @param {RTRemoteProtocol} protocol the protocol instance
   * @param {object} message the message entity
   */
  constructor(protocol, message) {
    this.protocol = protocol;
    this.message = message;
  }
}


module.exports = RTRemoteTask;
